### ISSUE TITLE
Cannot find how to configure the API key

### ISSUE THREAD

**Issue by newfulluser** [barnumbirr/coinmarketcap#47]
The wrapper seems to be working fine without one, but I don't want to get my IP rate limited. In the future I may have to purchase one of the paid api plans to avoid getting limited.

I have looked around and couldn't find where am I supposed to type my api key.

**Comment by alex-stout** [comment]
I don't think this has been looked into yet. I'll check out what would be needed to make the conversion. As of now, this is not supported.

**Comment by barnumbirr** [comment]
Ohai,

haven't found the time to update yet, I'll try my best to do so before the deadline.

Cheers.

**Comment by alex-stout** [comment]
@dahifi I'll take some time this weekend to look at refactoring the code for the new calls.

**Comment by webventurer** [comment]
CoinMarketCap are discontinuing the v2 API on 04.12.18 btw:
https://coinmarketcap.com/api/

**Comment by tmlee** [comment]
If you guys are looking for a public open API for market data and more, do consider our API at https://www.coingecko.com/api
We designed it to be easy to consume and I think you guys might find it interesting. 
Happy to answer any questions or concerns you may have!

**Comment by newfulluser** [comment]
It seems to be working but probably not for long.

**Comment by plgonzalezrx8** [comment]
Any idea if this will be updated to be able to use the new APIs? I don't want to lose the functionality of my setup. 

Owner, not sure if you are reading this, but will you continue to maintain this library? You are in a very priviledged position of having the default library when doing the pip install coinmarketcap.

**Comment by kodycode** [comment]
I started updating the functions to the new API on my own fork. Right now I have the wrapper using the coinmarketcap sandbox API on one branch, but will update it to use the pro version later tonight or tomorrow. See here for the sandbox branch: https://github.com/kodycode/coinmarketcap/tree/sandbox_branch

I don't think I'll be covering the other endpoints but just the main ones.

**Comment by xmatthias** [comment]
@kodycode will your pr require changes from a users perspective, or would an existing implementation continue to work as is?

**Comment by kodycode** [comment]
@xmatthias Yes, it would require changes from a users perspective. The first change of this updated implementation you would need to supply an api key as one of the arguments for creating the `Market` class. Not only that, but the functions don't return the same data they used to (see [here](https://coinmarketcap.com/api/documentation/v1/#operation/getV1CryptocurrencyListingsLatest) to see the data returned for the `listings` function, [here](https://coinmarketcap.com/api/documentation/v1/#operation/getV1CryptocurrencyQuotesLatest) for the `ticker` function, and [here](https://coinmarketcap.com/api/documentation/v1/#operation/getV1GlobalmetricsQuotesLatest) for the `stats` function).

There are also new arguments to feed to the endpoints if so desired. I've documented them in the comments of these functions in my branch, but you may also read them up on the links I posted as well.
